src/server/session.ts

```diff
@@ -181,18 +181,29 @@ module ts.server {
         }
 
         semanticCheck(file: string, project: Project) {
-            var diags = project.compilerService.languageService.getSemanticDiagnostics(file);
-            if (diags) {
-                var bakedDiags = diags.map((diag) => formatDiag(file, project, diag));
-                this.event({ file: file, diagnostics: bakedDiags }, "semanticDiag");
+            try {
+                var diags = project.compilerService.languageService.getSemanticDiagnostics(file);
+
+                if (diags) {
+                    var bakedDiags = diags.map((diag) => formatDiag(file, project, diag));
+                    this.event({ file: file, diagnostics: bakedDiags }, "semanticDiag");
+                }
+            }
+            catch (err) {
+                this.logError(err, "semantic check");
             }
         }
 
         syntacticCheck(file: string, project: Project) {
-            var diags = project.compilerService.languageService.getSyntacticDiagnostics(file);
-            if (diags) {
-                var bakedDiags = diags.map((diag) => formatDiag(file, project, diag));
-                this.event({ file: file, diagnostics: bakedDiags }, "syntaxDiag");
+            try {
+                var diags = project.compilerService.languageService.getSyntacticDiagnostics(file);
+                if (diags) {
+                    var bakedDiags = diags.map((diag) => formatDiag(file, project, diag));
+                    this.event({ file: file, diagnostics: bakedDiags }, "syntaxDiag");

```

Comment:
Condense `file: file` into a shorthand property assignment. If you renamed `diags` to `diagnostics`, you could do the same:

```TypeScript
this.event({ file, diagnostics }, "syntaxDiag")
```